armory/utils/metrics.py

```diff
@@ -151,6 +157,44 @@ def _snr_spectrogram(x_i, x_adv_i):
     return signal_power / noise_power
 
 
+@contextmanager
+def resource_context(name="Name", profiler="Basic"):
+    profiler_types = ["Basic", "Deterministic"]
+    if profiler not in profiler_types:
+        raise ValueError(f"Profiler {profiler} is not one of {profiler_types}.")
+    if profiler == "Deterministic":
+        logger.warn(
+            "Using Deterministic profiler. This may reduce timing accuracy and result in a large results file."
+        )
+        pr = cProfile.Profile()
+        pr.enable()
+    startTime = time.perf_counter()
+    yield
+    elapsedTime = time.perf_counter() - startTime
+    if profiler == "Deterministic":
+        pr.disable()
+        s = io.StringIO()
+        sortby = "cumulative"
+        ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
+        ps.print_stats()
```

Comment:
I would recommend removing the print line. It outputs a lot to the screen, which makes it difficult to see the other logs.